### PR TITLE
Moved error messages to translation file

### PR DESCRIPTION
**Pull request by nileshgulia1**
Related issue #878 

Error messages are moved to en.yml file.Tests are also passing.

### CODE REVIEW

File: .env-example [hunk 1]
```diff
@@ -1,9 +0,0 @@
-MAILTRAP_USER=InboxUsername
-MAILTRAP_PASSWORD=InboxPassword
-
-# Camo proxies insecure images in students' status updates
-CAMO_HOST=https://rgsoc-teams-camo.herokuapp.com
-CAMO_KEY=supersecret
-
-# Google Maps API key for e.g. city autocomplete
-GOOGLE_MAPS_API_KEY=YourGoogleMapsAPIKey
```
Comment (F3PiX) [inline .env-example]: We don't want to remove this file from the repo. (I don't think you need to remove it locally, either 🤔 ).
Anyway, removing it should not be part of this PR.

File: config/locales/en.yml [hunk 2]
```diff
@@ -89,3 +89,58 @@ en:
...
+                  attributes:
+                      projects:
+                          multiple_sel_error: must not be selected twice
+
+    errors:                      
+        models:
+          application_draft:
+                  attributes:
+                      projects:
+                          must_accepted: must have been accepted
```
[the remark this answers is not in the document]
Comment (F3PiX) [reply config/locales/en.yml]: Also, can you follow the same indentation as the rest of the file? It seems not to be super consistent, with sometimes 4 and sometimes 2 spaces. At first sight, it looks like it is easy to copy the same indentations.

File: config/database.yml.example [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-test:
-  adapter: postgresql
-  database: rgsocteams_test
-  host: localhost
-  min_messages: error
-
-production:
-  adapter: postgresql
-  database: rgsocteams_production
-  host: localhost
```
Comment (F3PiX) [inline config/database.yml.example]: Same as for the other removed file: we don't want this file removed. Can you leave the removal out of this PR please?

File: app/models/user.rb [hunk 4]
```diff
@@ -220,7 +220,7 @@ def complete_from_github
 
   def immutable_github_handle
     return if new_record?
-    errors.add(:github_handle, "can't be changed") if changes_include?(:github_handle)
+    errors.add(:github_handle, :can't_be_changed) if changes_include?(:github_handle)
   end
```
Comment (F3PiX) [inline app/models/user.rb]: Remove this whiteline please.

File: config/locales/en.yml [hunk 5]
```diff
@@ -89,3 +89,58 @@ en:
...
+                        no_more_than_two: there cannot be more than 2 students on a team.
+
+       errors:                      
+        models:
+           user:
+             attributes:
+                     github_handle:
+                             can't_be_changed: can't be changed
+
+
```
Comment (F3PiX) [inline config/locales/en.yml]: And remove the redundant whitelines please

File: .env-example (hunk 1 above)
Comment (nileshgulia1) [reply .env-example]: yes,sorry for that,the file got accidentally removed ,will add it back now :)

File: app/models/application_draft.rb [hunk 6]
```diff
@@ -173,19 +173,19 @@ def ready?
...
     if project1 && project1 == project2
-      errors.add(:projects, 'must not be selected twice')
+      errors.add(:projects, :multiple_sel_error)
     end
   end
 
   def accepted_projects_required
     if projects.any? { |p| p && !p.accepted? } # if they don't exist, the presence validation will handle it
-      errors.add(:projects, 'must have been accepted')
+      errors.add(:projects, :must_accepted)
```
[the remark this answers is not in the document]
Comment (nileshgulia1) [reply app/models/application_draft.rb]: yeah,sure :+1:

File: config/database.yml.example (hunk 3 above)
Comment (nileshgulia1) [reply config/database.yml.example]: yeah,sure :)

File: config/locales/en.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nileshgulia1) [reply config/locales/en.yml]: Thanks for being patient.I need another help :)
```errors:
        models:
          todo:
          application_draft:
          conference:
          team:
          user:
            attributes:
              user:
                too_many_reviewers: too many reviewers
              projects:
                too_many_selected: must not be selected twice
                none_accepted: must have been accepted
              base:
                too_many_applications: Only one application may be lodged
                unconfirmed_email_address: Please make sure every student confirmed the email address.
              ends_on:
                chronology_violation: must be a later date than start date
              roles:
                too_many_students: there cannot be more than 2 students on a team.
              github_handle:
                immutabilty_violation: can't be changed
```

 What's wrong with this code, my local tests are failing?

[the remark this answers is not in the document]
Comment (F3PiX) [reply config/locales/en.yml]: No worries :-) We've all been there! I really appreciate your effort. 

Take a look at this: http://docs.ansible.com/ansible/latest/YAMLSyntax.html 
You should collect the attributes for each key. 
Like this:
```
errors:  
    models:  
        todo:   
            attributes:
                user:
                    too_many_reviewers: too many reviewers
        application_draft:
              attributes:
                  projects:
                      must_accepted: must have been accepted
                      # other error messages for projects here
                  #other_attribute:
                      #error messages on other_attribute 
       #yet another model:
              attributes:
                   #etc
 
```
Does that make sense?

[the remark this answers is not in the document]
Comment (nileshgulia1) [reply config/locales/en.yml]: Yes,exactly I have already written the above code you mentioned earlier, but this is not showing in the changes .please check my latest commit 9dba120

File: config/locales/en.yml [hunk 7]
```diff
@@ -89,3 +89,24 @@ en:
...
+                too_many_applications: Only one application may be lodged
+                unconfirmed_email_address: Please make sure every student confirmed the email address.
+          conference:
+            attributes:
+              ends_on:
+                chronology_violation: must be a later date than start date
+          team:
+            attributes:
+              roles:
+                too_many_students: there cannot be more than 2 students on a team.
```
Comment (F3PiX) [inline config/locales/en.yml]: For the team, two base errors are missing here. (line 175 and 183)
https://github.com/rails-girls-summer-of-code/rgsoc-teams/blob/a020e98eb1339e29170fdb7e7c64404fe0719b2c/app/models/team.rb#L165-L184

It is a bit more challenging than the other errors. Are you up to that?

**Comment by nileshgulia1** [comment]
@F3PiX  thanks for the help so far,I will close the PR for now and will submit a clean PR soon :)
